parquet/src/test/java/com/netflix/iceberg/parquet/TestParquetMetrics.java

```diff
@@ -0,0 +1,214 @@
+    checkFieldMetrics(4, schema, metrics, 2L, 0L, 1.0F, 2.0F);
+    checkFieldMetrics(5, schema, metrics, 2L, 1L, 2.0D, 2.0D);
+    checkFieldMetrics(6, schema, metrics, 2L, 1L, new BigDecimal("3.50"), new BigDecimal("3.50"));
+    checkFieldMetrics(7, schema, metrics, 2L, 0L, CharBuffer.wrap("AAA"), CharBuffer.wrap("ZZZ"));
+    checkFieldMetrics(8, schema, metrics, 2L, 1L, 1500, 1500);
+    checkFieldMetrics(9, schema, metrics, 2L, 0L, 2000L, 3000L);
+    checkFieldMetrics(10, schema, metrics, 2L, 0L, 0L, 1000L);
+    // TODO: enable once issue#126 is resolved
+    // checkFieldMetrics(11, schema, metrics, 2L, 1L, uuid, uuid);
+    checkFieldMetrics(12, schema, metrics, 2L,
+      0L, ByteBuffer.wrap(fixed.bytes()), ByteBuffer.wrap(fixed.bytes()));
+    checkFieldMetrics(13, schema, metrics,
+      2L, 0L, ByteBuffer.wrap("S".getBytes()), ByteBuffer.wrap("W".getBytes()));
+  }
+
+  @Test
+  public void testMetricsForListAndMapElements() throws IOException {
+    StructType structType = StructType.of(
+      required(1, "leafIntCol", IntegerType.get()),
+      optional(2, "leafStringCol", StringType.get())
+    );
+    Schema schema = new Schema(
+      optional(3, "intListCol", ListType.ofRequired(4, IntegerType.get())),
+      optional(5, "mapCol", MapType.ofRequired(6, 7, StringType.get(), structType))
+    );
+
+    Record record = new Record(AvroSchemaUtil.convert(schema.asStruct()));
+    record.put("intListCol", Lists.newArrayList(10, 11, 12));
+    Record struct = new Record(AvroSchemaUtil.convert(structType));
+    struct.put("leafIntCol", 1);
+    struct.put("leafStringCol", "BBB");
+    Map<String, Record> map = Maps.newHashMap();
+    map.put("4", struct);
+    record.put(1, map);
+
+    File parquetFile = writeRecords(schema, record);
+
+    Metrics metrics = ParquetMetrics.fromInputFile(localInput(parquetFile));
+    Assert.assertEquals(1L, (long) metrics.recordCount());
+    checkFieldMetrics(1, schema, metrics, 1, 0, null, null);
+    checkFieldMetrics(2, schema, metrics, 1, 0, null, null);
+    checkFieldMetrics(4, schema, metrics, 3, 0, null, null);
+    checkFieldMetrics(6, schema, metrics, 1, 0, null, null);
+  }
+
+  @Test
+  public void testMetricsForNullColumns() throws IOException {
+    Schema schema = new Schema(
+      optional(1, "intCol", IntegerType.get())
+    );
+    Record firstRecord = new Record(AvroSchemaUtil.convert(schema.asStruct()));
+    firstRecord.put("intCol", null);
+    Record secondRecord = new Record(AvroSchemaUtil.convert(schema.asStruct()));
+    secondRecord.put("intCol", null);
+
+    File parquetFile = writeRecords(schema, firstRecord, secondRecord);
+
+    Metrics metrics = ParquetMetrics.fromInputFile(localInput(parquetFile));
+    Assert.assertEquals(2L, (long) metrics.recordCount());
+    checkFieldMetrics(1, schema, metrics, 2, 2, null, null);
+  }
+
+  private <T> void checkFieldMetrics(int fieldId, Schema schema, Metrics metrics,
```

Comment:
I think these tests would be more readable if this were broken into two: `assertCounts` and `assertBounds`. That way it is clear when calling them that the metrics match the value and null counts, without looking at the implementation. Similarly, it would be clear that the bounds are equal to the given values.

That also allows you to avoid passing in so many arguments. The schema becomes unnecessary because you pass in the field or field type and field ID. As it is now, this gets the field by ID and then accesses its field ID, which is awkward.